README.md

```diff
@@ -184,14 +189,149 @@ Binaries built by Burrito include a built-in set of commands for performing main
+  * `Build` -  This is the final phase in the build flow, it produces the final wrapper binary with a payload embedded inside.
+  
+  You can add your own steps before and after phases execute. Your custom steps will also receive the build context struct, and can return a modified one to customize a build to your liking.
+
+  An example of adding a step before the fetch phase, and after the build phase:
+
+  ```elixir
+  # ... mix.exs file
+  def releases do
+    [
+      my_app: [
+        steps: [:assemble, &Burrito.wrap/1],
+        burrito: [
+          # ... other Burrito configuration
+          extra_steps: [
+            fetch: [pre: [MyCustomStepModule, AnotherCustomStepModule]],
+            build: [post: [CustomStepAgain, YetAnotherCustomStepModule]]
+            # ...
+          ]
+        ]
+      ]
+    ]
+  end
+  # ...
+  ```
+
+#### Build Targets and Qualifiers
+
+A burrito build target is a keyword list that contains an operating system, a CPU architecture, and extra build options (called Qualifiers).
+
+Here's a build target that defines Linux, 64-bit x86:
+
+```elixir
+  def releases do
+  [
+    example_cli_app: [
+      steps: [:assemble, &Burrito.wrap/1],
+      burrito: [
+        targets: [
+          linux: [os: :linux, cpu: :x86_64],
+        ],
+      ]
+    ]
+  ]
+  end
+```
+
+If we want to define another build target for linux, but with Musl libc instead of GNU libc we can add a Build Qualifier.
+
+```elixir
+  def releases do
+  [
+    example_cli_app: [
+      steps: [:assemble, &Burrito.wrap/1],
+      burrito: [
+        targets: [
+          linux: [os: :linux, cpu: :x86_64],
+          # add a musl libc linux target below...
+          linux_musl: [os: :linux, cpu: :x86_64, libc: :musl],
+        ],
+      ]
+    ]
+  ]
+  end
+```
+
+Build qualifiers are a simple way to pass specific flags into the Burrito build pipeline. Currently, only the `libc` and `local_erts` qualifiers have any affect on the standard burrito build phases and steps.
+
+Tip: You can use these qualifiers as a way to pass per-target information into your custom build steps.
+
+#### Using Custom ERTS Builds
+
+The Burrito project provides pre-compiled builds of Erlang for the following platforms:
+
+```elixir
+[os: :darwin, cpu: :x86_64],
+[os: :linux, cpu: :x86_64, libc: :gnu], # or just [os: :linux, cpu: :x86_64]
+[os: :linux, cpu: :x86_64, libc: :musl],
+[os: :windows, cpu: :x86_64]
+```
+
+If you require another build of the ERTS for your targets, and wish to provide one yourself, you can do so by specifying the `local_erts` build qualifier for a specific target.
+
+Example:
+
+```elixir
+  def releases do
+  [
+    example_cli_app: [
+      steps: [:assemble, &Burrito.wrap/1],
+      burrito: [
+        targets: [
+          linux_arm: [os: :linux, cpu: :arm64, local_erts: "/path/to/my_custom_erts.tar.gz"]
```

Comment:
Longer examples of build targets may be clearer to the reader if the options are broken up onto their own lines. Doing so also makes it easier to copy smaller fragments of the sample:

```elixir
targets: [
  linux_arm: [
    os: :linux,
    cpu: :arm64,
    local_erts: "/path/to/my_custom_erts.tar.gz"
  ]
]
```